controller/AuthoringTool.php

```diff
@@ -87,4 +99,52 @@ public function launch(): void
 
         $this->forward('run', null, null, $_GET);
     }
+
+    /**
+     * @return bool
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    private function isFeatureTaoAsToolEnabled(): bool
+    {
+        return $this->getServiceManager()
+            ->getContainer()
+            ->get(FeatureFlagChecker::class)
+            ->isEnabled(FeatureFlagCheckerInterface::FEATURE_FLAG_TAO_AS_A_TOOL);
+    }
+
+    /**
+     * @return LtiMessagePayloadInterface
+     * @throws ContainerExceptionInterface
+     * @throws InterruptedActionException
+     * @throws LtiException
+     * @throws NotFoundExceptionInterface
+     */
+    private function getLtiMessageOrRedirectToLogin(): LtiMessagePayloadInterface
+    {
+        if (!$this->isFeatureTaoAsToolEnabled()) {
+            $this->getLogger()->info(
+                'TAO as tool feature is disabled. The user will be redirected to the login page.'
+            );
+            $this->redirect(_url('login', 'Main', 'tao'));
```

Comment:
I would separate this business logic from that method. IMO it is violating Single Responsibility Principle

Reply:
Which responsibility do you mean? It is the responsibility of a controller layer to do redirects

Reply:
it either geting the LtiMessage or it redirect to login

Reply:
If you are talking about the method itself then, in this case, it loses its proposes so I can just move this logic back to the controller. Honestly, I do not like the idea as the method explains exactly what he is supposed to do. Asisde from that the Single Responsability principle applies to the class itself, not methods.

Reply:
If I had 2 redirects like in your example, I would just rename it to `getLtiMessageOrPerformRedirect` that would be the simplest way.